src/corehost/cli/corehost_context_contract.h

```diff
@@ -66,4 +67,16 @@ static_assert(offsetof(corehost_context_contract, run_app) == 5 * sizeof(size_t)
 static_assert(offsetof(corehost_context_contract, get_runtime_delegate) == 6 * sizeof(size_t), "Struct offset breaks backwards compatibility");
 #pragma pack(pop)
 
+/// Signature of delegate returned by coreclr_delegate_type::load_assembly_and_get_function_pointer
+typedef int (STDMETHODCALLTYPE *LoadAssemblyAndGetFunctionPointer)(
+    const pal::char_t * assemblyPathNative /* Fully qualified path to assembly */,
+    const pal::char_t * typeNameNative     /* Assembly qualified type name */,
+    const pal::char_t * methodNameNative   /* Public static method name compatible with delegateType */,
+    const pal::char_t * delegateTypeNative /* Assembly qualified delegate type name or null */,
+    void *              reserved           /* Extensibility parameter (currently unused and must be 0) */,
+    /*out*/ void **     delegate           /* Pointer where to store the function pointer result */);
+
+/// Signature of delegate returned by LoadAssemblyAndGetFunctionPointer when delegateTypeNative == null (default)
+typedef int (__cdecl *ComponentEntryPointDelegate)(void *arg, int32_t argSize);
```

Comment:
It seems weird that we would make the default delegate type `cdecl` when delegates and exports from `coreclr` are all `stdcall`. Thoughts on changing it to `stdcall`, so the split kind of stays things out of `coreclr` = `stdcall`, and  `hostfxr`/`hostpolicy` = `cdecl` - @AaronRobinsonMSFT?

Reply:
Chatted offline. Changing to `stdcall`